### ISSUE TITLE
Add support for Lightning Web Components Open Source

### ISSUE THREAD

**Issue by muenzpraeger** [forcedotcom/salesforcedx-vscode#1599]
**Is your feature request related to a problem? Please describe.**
The LWC extension does currently not support projects that are based on LWC OSS using `lwc-create-app`.

**Describe the solution you'd like**
New menu items should be shipped that augment the implementation of LWC OSS in [lightning-language-server](https://github.com/forcedotcom/lightning-language-server).

https://github.com/forcedotcom/lightning-language-server/issues/15

The following additions are proposed:
- New context menu "LWC: Create Lightning Web Component" in file explorer on `modules` folder
 - New command palette item "LWC: Create Lightning Web Component"

Optional:
- Adding default tasks for watch/build etc. modes.

**Comment by muenzpraeger** [comment]
@ntotten Good with this? I would then start adding them. ;-)

**Comment by muenzpraeger** [comment]
LOL - I forgot a similar issue that I created myself before... 😂🙄

**Comment by muenzpraeger** [comment]
@lcampos I'm currently working on the implementation. As I don't have any deps to the Salesforce CLI I am wondering what your preferred way would be to create the LWC files.

- Option 1: I store the template files within this project and copy them over.
- Option 2: I store the template files in `lwc-services`, add a command to it, and then invoke it from the extension.

Personally I think option 2 is overkill, especially as things don't really change on this front.

Thougts?

**Comment by lcampos** [comment]
@muenzpraeger can you expand a bit more on the LWC files you are looking at creating ? Do you want to be able to create the bundles without the `-meta.xml` file that's currently created by the template commands or are we talking of different things ?

**Comment by muenzpraeger** [comment]
Exactly that, @lcampos. Same files, no metadata.

**Comment by lcampos** [comment]
Ok, we already have that in the current CLI commands and the VSCode extensions. In the extensions you can add the `"salesforcedx-vscode-core.internal-development": true,` and directly on the CLI you just add the `--internal` flag to the commands.

**Comment by muenzpraeger** [comment]
Yeah, but using Salesforce CLI shouldn’t be a precondition.

**Comment by lcampos** [comment]
Sure, but if we start adding templates directly to the extensions then we'll now have two different places to host the same templates.

We are working towards having an easier way of loading the template plugin directly in the extensions. That is part of the whole refactoring effort we've gone through with that plugin. Basically building a module from the same repo that is used by the cli that can also be directly consumed by the extensions (similar to what we do with `@salesforce/core`) but we're not there yet.

**Comment by AnanyaJha** [comment]
Hi @muenzpraeger! In the past year we've made the transition to a library-based architecture so the CLI and the Extensions will share the same business logic - and you won't need to have the CLI installed to work with the Extensions. We're working on making this change across all the Extensions features, but so far we've addressed the templates for LWC creation. Let me know if you were looking for something more with this issue, else I'll close this soon. Thanks for the feedback, feel free to keep dropping comments :)
